apps/src/feedback.js

```diff
@@ -565,8 +565,12 @@ FeedbackUtils.prototype.getFeedbackMessage_ = function(options) {
           puzzleNumber: options.level.puzzle_number || 0
         };
         if (options.feedbackType === TestResults.FREE_PLAY && !options.level.disableSharing) {
-          message = finalLevel ? msg.finalStage(msgParams) : '';
-          message = message + ' ' + options.appStrings.reinfFeedbackMsg;
+          if (options.level.disableFinalStageMessage) {
+            message = options.appStrings.reinfFeedbackMsg;
+          } else {
+            message = finalLevel ? msg.finalStage(msgParams) : '';
+            message = message + ' ' + options.appStrings.reinfFeedbackMsg;
```

Comment:
Some other places in this file you defend against null appStrings but not here, should we add that?

Do you want the leading space even when message is ''?